src/dialog.jsx

```diff
@@ -379,27 +375,30 @@ let Dialog = React.createClass({
     }
   },
 
-  _onShow() {
-    if (this.props.onShow) this.props.onShow();
+  _show() {
+    this.refs.dialogOverlay.preventScrolling();
+    this.setState({ open: true });
+  },
+
+  _dismiss() {
+    CssEvent.onTransitionEnd(ReactDOM.findDOMNode(this), () => {
+      this.refs.dialogOverlay.allowScrolling();
+    });
+
+    this.setState({ open: false });
   },
 
-  _onDismiss() {
-    if (this.props.onDismiss) this.props.onDismiss();
+  _requestClose() {
+    if (this.props.onRequestClose) this.props.onRequestClose();
```

Comment:
If we no longer trigger `onRequestClose` when the Dialog is uncontrolled we would have to re-add the `onDismiss` callback since we otherwise would not know if a dialog has been closed.
This could then be confusing as to when `onDismiss` should be triggered and when not. For example: should it be triggered when we explicitly change `isOpen` to false?